src/coreclr/vm/threadsuspend.cpp

```diff
@@ -4600,6 +4605,9 @@ void Thread::HijackThread(ExecutionState *esb X86_ARG(ReturnKind returnKind) X86
 
     // Remember the place that the return would have gone
     m_pvHJRetAddr = *esb->m_ppvRetAddrPtr;
+#if defined(TARGET_ARM64)
+    m_pvHJRetAddr = PacStripPtr(m_pvHJRetAddr);
```

Comment:
Same as my other comment: Avoid storing the wrong value and overwriting it with the correct value.